src/main/java/com/teamscale/upload/xcode/XCResultConverter.java

```diff
@@ -164,6 +164,9 @@ private List<File> convertToXccovArchives(File reportDirectory, ActionsInvocatio
 
 		for (int i = 0; i < actionsInvocationRecord.actions.length; i++) {
 			ActionRecord action = actionsInvocationRecord.actions[i];
+			if(action == null){
```

Comment:
It's not quite clear if the `action` is the one that causes the NPE. Could also be `action.actionResult.coverage`. Please add null checks for all of the nested fields.